redbot/cogs/warnings/warnings.py

```diff
@@ -57,10 +58,31 @@ async def red_delete_data_for_user(
 
         all_members = await self.config.all_members()
 
-        async for guild_id, guild_data in AsyncIter(all_members.items(), steps=100):
-            if user_id in guild_data:
+        c = 0
+
+        for guild_id, guild_data in all_members.items():
+            c += 1
+            if not c % 100:
+                await asyncio.sleep(0)
+
+            if str(user_id) in guild_data:
```

Comment:
casting to int is done by `config.all_members()`
```suggestion
            if user_id in guild_data:
```

Reply:
*sigh* had that one right before, and missed that on revisiting.